rust/kernel/clk.rs

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Clock subsystem
+//!
+//! C header: [`include/linux/clk.h`](../../../../include/linux/clk.h)
+
+use crate::{
+    bindings,
+    device::Device,
+    error::{from_kernel_err_ptr, Error, Result},
+    str::CStr,
+};
+
+/// Represents `struct clk *`
+pub struct Clk(*mut bindings::clk);
+
+impl Clk {
+    /// returns value of rate field of `struct clk`
+    pub fn get_rate(&self) -> u32 {
+        // SAFETY: self object contains aquired, valid pointer to `struct clk`
+        unsafe { bindings::clk_get_rate((*self).0) }
+    }
+}
+
+/// clk_prepare_enable helps cases using clk_enable in non-atomic context
```

Comment:
While this is from the original docs, it would be best to try to adapt the documentation to at least Markdown. In this case, we only have `prepare_enable` so far, so I would simply say:

```suggestion
/// Do not use in atomic context.
```

Reply:
ack, thanks